src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1984,6 +2087,8 @@ impl Cage {
                 //incrementing the ref count so that when close is executed on the dup'd file
                 //the original file does not get a negative ref count
                 match *inodeobj {
+                    // increments the reference count of the file object associated with the original file descriptor
+                    // to ensure that the file object is not deleted until all its associated file descriptors are closed.  
```

Comment:
Not able to add the comment in the lines below, but can you explain why are we using the `O_CLOEXEC` flag while updating the flags for all FileDesc types?

Reply:
when a fd is duplicated we don't want the new fd to get the use the old 0_CLOEXEC